src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -1327,9 +1351,19 @@ static int client_receive_message(
 }
 
 int sd_dhcp6_client_stop(sd_dhcp6_client *client) {
+        int r;
+
         if (!client)
                 return 0;
 
+        if (client->send_release) {
```

Comment:
I guess the condition should be improved.
- The client has lease,
- The lease has at lease one valid IA_NA or IA_PD.